File: src/2520.count-the-digits-that-divide-a-number.2608/solution.ts
function countDigits (num: number): number {
    let copy = num;
    let result = 0;
    while (copy) {
        const digit = copy % 10;
        if (num % digit === 0) {
            result++;
        }
        copy = (copy - digit) / 10;
    }
    return result;
}

File: src/2520.count-the-digits-that-divide-a-number.2608/solution.java
class Solution {
    public int countDigits(int num) {
        int[] counts = new int[10];
        int copy = num;
        while (copy != 0) {
            counts[copy % 10]++;
            copy = copy / 10;
        }
        int result = 0;
        for (int i = 1; i < 10; i++) {
            if (counts[i] == 0) {
                continue;
            }
            if (num % i == 0) {
                result += counts[i];
            }
        }
        return result;
    }
}